tsconfig.json

```diff
@@ -34,8 +34,12 @@
     "types": []
   },
   "include": [
-    "**/docs/types/*.d.ts",
-    "**/@stdlib/types/*.d.ts"
+    "docs/types/index.d.ts",
```

Comment:
If this file is moved to `/etc`, would these glob patterns need to be updated?

Reply:
Probably. Is there a need to move this file though?

I hope not; getting the `include` entries in this file working was, one may not believe it, actually one of the most time-consuming things for me while drafting this PR. The VSCode ESLint extension has some undocumented resolution behavior that by default will exclude files in `node_modules` folders (so including our `lib/node_modules` ones), and I had to add more specific glob patterns to get files to lint properly. 

For the added entries at least, we cannot have starting `**` and need the leading `lib/node_modules` to work in VSCode. Would be good to make sure things work as-is in Sublime Text too.

Reply:
> Is there a need to move this file though?

It is not a hard requirement. If possible, my preference would be in `/etc` as this provides a single location for all config files and minimizes the cluttering of the top-level directory.

If it is not possible due to editor issues, then we don't have much of a choice.